third_party/terraform/tests/resource_iam_beta_workload_identity_pool_provider_test.go.erb

```diff
@@ -0,0 +1,198 @@
+<% autogen_exception -%>
+package google
+
+<% unless version == 'ga' %>
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccIAMBetaWorkloadIdentityPoolProvider_aws(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"random_suffix": randString(t, 10),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
```

Comment:
You'll additionally want to add a `CheckDestroy` to each test case, similar to [`auto-pr-4129-old..auto-pr-4129`#diff-3a6f1c9a16](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-4129-old..auto-pr-4129#diff-3a6f1c9a16ef949d7eebb064e5c2cd641cfdda35e77cb166616646dc02aed7b8R79) (it will call the same function, since it's the same resource)

Reply:
I have made the changes to treat a DELETED state as if the resource is gone. I understand what the intention is, but I still don't see any affect on the tests. Without these changes I would have expected the tests to fail since they would still find the deleted resource at the end of the test. However, the tests where green.
Even with these new changes if I remove the `if v := res["state"]; v == "DELETED" { return nil }` code from the custom_check_destroy, the tests still succeed. When enabling TF logging I can see there is no GET being done to see if the resources still exist, so this code remains untested.

When I intentionally throw an error during the check everything still succeeds:
```
func testAccCheckIAMBetaWorkloadIdentityPoolDestroyProducer(t *testing.T) func(s *terraform.State) error {
	return func(s *terraform.State) error {
		for name, rs := range s.RootModule().Resources {
			return fmt.Errorf("IAMBetaWorkloadIdentityPool still exists at %s", rs.Type)
			if rs.Type != "google_iam_workload_identity_pool" {
				continue
			}
```

Only when I throw the error outside of the for loop, the tests will fail. This makes me think that `s.RootModule().Resources` is not giving any resources to check the existence for.

Reply:
Interesting! I'm running the tests, I'll see what I find.

Reply:
I looked into this briefly, and found the same thing in another resource! This loop is not being entered- I suspect it's a general SDK issue. Not a blocker for this PR imo.

Reply:
Found it! https://github.com/hashicorp/terraform-plugin-sdk/blob/master/CHANGELOG.md#204-october-06-2020


We're using `2.0.3`, so I'll bump the providers.